integrationTests/vm/systemVM/systemVm_test.go

```diff
@@ -75,9 +75,10 @@ func TestStakingUnstakingAndUnboundingOnMultiShardEnvironment(t *testing.T) {
 	nonce++
 
 	///////////------- send stake tx and check sender's balance
+	var txData string
```

Comment:
this test is too long >150 lines of code, you can break this test into several functions you can call.
there is also duplicated code for
```go
for i := 0; i < nrRoundsToPropagateMultiShard; i++ {
	integrationTests.ProposeBlock(nodes, idxProposers, round, nonce)
	integrationTests.SyncBlock(t, nodes, idxProposers, round)
	round = integrationTests.IncrementAndPrintRound(round)
	nonce++
}
```
 this can be extracted into a function and call that instead every time needed

then every section of code marked by a comment, could again be extracted into a function:
- verify initial balance
- send stake
- check accounts after staking
- send unstake
- send unbound
- verify unbound

maybe also add some verification for unstake

Reply:
done.